core/src/main/java/com/sequenceiq/cloudbreak/service/multiaz/MultiAzCalculatorService.java

```diff
@@ -36,9 +41,11 @@
 @Service
 public class MultiAzCalculatorService {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(MultiAzCalculatorService.class);
+
     private static final String DEFAULT_RACK = "default-rack";
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(MultiAzCalculatorService.class);
+    private static final int ALLOWED_NUMBER_OF_HINTS = 3;
```

Comment:
Why do we allow only 3 different AZ?

Reply:
If the number of hints is smaller than 3, we will fall back to round-robin.